models/classes/Lists/Business/Service/RemoteSourceParserInterface.php

```diff
@@ -14,27 +15,21 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  *
- * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
- *
+ * Copyright (c) 2020-2021 (original work) Open Assessment Technologies SA;
  */
 
 declare(strict_types=1);
 
 namespace oat\tao\model\Lists\Business\Service;
 
-use oat\tao\model\Lists\Business\Domain\Value;
-use RuntimeException;
+use oat\tao\model\Context\ContextInterface;
 
 interface RemoteSourceParserInterface
 {
     /**
-     * @param array  $json
-     * @param string $uriRule
-     * @param string $labelRule
-     *
-     * @return iterable|Value[]
-     *
-     * @throws RuntimeException
+     * @deprecated Use $this->iterateContext()
      */
     public function iterate(array $json, string $uriRule, string $labelRule): iterable;
+
+    public function iterateContext(ContextInterface $context): iterable;
```

Comment:
You are not actually iterating the context, but based on it. Wdyt?

```suggestion
    public function iterateByContext(ContextInterface $context): iterable;
```